modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -1351,3 +1359,92 @@ func (p *ProjectPipelineService) ListUsedRefs(ctx context.Context, params deftyp
 	}
 	return resp.Ref, nil
 }
+
+type pipelineCategoryRule struct {
+	Key        string
+	Category   string
+	Rules      []string
+	RunningNum uint64
+	FailedNum  uint64
+	TotalNum   uint64
```

Comment:
you can separate these fields as a common struct like `PipelineStatisticsNums`

Reply:
done